src/inmanta/execute/scheduler.py

```diff
@@ -269,6 +269,12 @@ def define_types(self, compiler: "Compiler", statements: Sequence["Statement"],
             ),
         ):
             t.normalize()
+            if isinstance(t, plugins.Plugin):
+                LOGGER.debug(
+                    "Inmanta type signature inferred from type annotations for plugin %s::%s",
+                    t.namespace,
+                    t.get_signature(dsl_types=True),
+                )
```

Comment:
Wouldn't it be cleaner to do this in `Plugin.normalize()`?

Reply:
yes

Reply:
```suggestion
            if isinstance(t, plugins.Plugin):
                LOGGER.debug(
                    "Found plugin %s::%s",
                    t.namespace,
                    t.get_signature(dsl_types=True),
                )
```

i.e. it may or may not be inferred